Tools/Cli-LoRa-Device-Provisioning/ValidationHelper.cs

```diff
@@ -0,0 +1,445 @@
+
+            foreach (string dr in euValidDataranges)
+                result += dr + ", ";
+
+            result += ".";
+
+            result = result.Substring(0, result.Length - 2);
+            return result;
+        }
+
+        public static string GetUSDataRates()
+        {
+            string result = "US: ";
+
+            foreach (string dr in usValidDataranges)
+                result += dr + ", ";
+
+            result = result.Substring(0, result.Length - 2);
+            result += ".";
+
+            return result;
+        }
+
+        public static string CleanString(string inString)
+        {
+            string outString = null;
+
+            if (!string.IsNullOrEmpty(inString))
+            {
+                outString = inString.Replace("\'", string.Empty);
+            }
+
+            return outString;
+        }
+
+        public static string CleanNetId(string inNetId)
+        {
+            string outNetId = null;
+
+            if (!string.IsNullOrEmpty(inNetId))
+            {
+                outNetId = inNetId.Replace("\'", string.Empty);
+                if (outNetId.Length < 6)
+                    outNetId = string.Concat(new string('0', 6).Substring(outNetId.Length), outNetId);
+
+                if (outNetId.Length > 6)
+                    outNetId = outNetId.Substring(outNetId.Length - 6);
+            }
+
+            return outNetId;
+        }
+
+        public static string GetTwinPropertyValue(dynamic value)
+        {
+            if (value == null)
+            {
+                return string.Empty;
+            }
+
+            if (value is string valueString)
+            {
+                return valueString.Trim();
+            }
+
+            if (value is bool valueBool)
+            {
+                if (valueBool)
+                    return "true";
+                else
+                    return "false";
+            }
+
+            if (value is uint valueUInt)
+            {
+                return valueUInt.ToString();
+            }
+
+            if (value is int valueInt)
+            {
+                return valueInt.ToString();
+            }
+
+            return value.ToString();
+        }
+
+        public static bool ValidateHexStringTwinProperty(string hexString, int byteCount, out string error)
+        {
+            error = string.Empty;
+
+            // hexString not dividable by 2.
+            if (hexString.Length % 2 > 0)
+            {
+                error = "Hex string must contain an even number of characters";
+                return false;
+            }
+
+            // hexString doesn't contain byteCount bytes.
+            if (hexString.Length / 2 != byteCount)
+            {
+                error = $"Hex string doesn't contain the expected number of {byteCount} bytes";
+                return false;
+            }
+
+            // Verify each individual byte for validity.
+            for (int i = 0; i < hexString.Length; i += 2)
```

Comment:
Ah, interesting. So it would be:
(int i = 0; i < hexString.Length && i+1 < hexString.Length; i += 2)
Should I put this and remove the previous check???

Reply:
if i+1 < hexString.Length then i < than hexString.Length will always be true right? So no need for the 2nd check correct?

Reply:
Yeah, absolutely right...!

Reply:
Changed it to your original suggestion. Double safety feature!!!